src/runtime-rs/crates/hypervisor/README.md

```diff
@@ -89,6 +92,215 @@ pub trait Hypervisor: Send + Sync {
     async fn save_state(&self) -> Result<HypervisorState>;
    }
 ```
+### VM start procedure: steps through time
 In current design, VM will be started in the following steps.
 
 ![vmm start](../../docs/images/vm-start.svg)
+
+### Semantic of each function of the trait
+In runtime-rs, most functions' implementation are VMM-specific, since currently `Dragonball` is default
+VMM and implemented correctly, following section will explain two things.
+1. In high-level, what is the expected behavior of each function.
+2. What `Dragonball`(DBS) does in each function. This could be a reference, but remember that each
+VMM should have their own orientations.
+
+#### Explanations
+```rust
+async fn prepare_vm(&self, id: &str, netns: Option<String>) -> Result<()>
+```
+1. What is expected?
+* In general, `prepare_vm()` need the VMM to prepare everything **that VMM needs before boot**. 
+For example, maybe fill in some fields of VMM, or maybe create something on host that the VMM needs when booting. But only some preparations need to be done.
+
+
+2. What DBS does?
+* DBS fills in some of its fields, e.g. `id, state, jailer_root, vm_path, netns`, and create a Unix Domain Socket(uds) for communication. Note that the uds is not directly added to the vmm, but added to a buffer. The devices
+in the buffer will be added after the VM starts.
+
+---
+
+```rust
+async fn start_vm(&self, timeout: i32) -> Result<()>
+```
+1. What is expected?
+* In general, `start_vm()` need the VMM and kernel to start and function correctly, which means ALL other functions from the trait `Hypervisor` should be able to return an expected result. This may be based on the preparation that `prepare_vm()` has done.
+
+2. What DBS does?
+* DBS first run its VMM server.
+    * It creates the jailer path of the VM, setup the channel for thread communication, and start a thread to run its event_loop, since DBS runs as the thread within the process. Other VMM may start another process.
+* DBS then starts its kernel.
+    * It does normal things as a VMM, prepare kernel_params, initrd, image, rootfs, etc. and runs with its kernel started.
+
+---
+
+```rust
+async fn stop_vm(&self) -> Result<()>
+```
+1. What is expected?
+* In general, `stop_vm()` wants you to exit the VMM from its execution.
+
+2. What DBS does?
+* It sends an stopVM request to DBS VMM server, the VMM server will receive the request and exit its execution from the thread called `vmm_thread`.
+
+---
+
+```rust
+    async fn pause_vm(&self) -> Result<()>;
+```
+1. What is expected?
+* In general, `pause_vm()` wants you to pause the VMM from its execution like it said. The ways to actually pause the VMM differs from VMMs.
+
+2. What DBS does?
+* Not implemented in DBS.
+
+---
+
+```rust
+    async fn save_vm(&self) -> Result<()>;
+```
+1. What is expected?
+* TBD
+
+2. What DBS does?
+* Not implemented in DBS.
+
+---
+
+```rust
+    async fn resume_vm(&self) -> Result<()>;
```

Comment:
It's better to move `resume_vm` to after `pause_vm`.

Reply:
Fixed.